esmvaltool/diag_scripts/radiation_budget/radiation_budget.py

```diff
@@ -0,0 +1,480 @@
+
+    Returns
+    -------
+    dictionary
+        The validated variable.
+
+    Examples
+    --------
+    >>> var1 = {"name": "sw_reflected_clouds", "unit": "W m-2", "data": 79.0}
+    >>> var2 = {"name": "toa_outgoing_longwave_flux", "unit": "W m-2",
+    ...         "data": 239.0}
+    >>> variable_data = [var1, var2]
+    >>> name = "sw_reflected_clouds"
+    >>> unit = "W m-2"
+    >>> validated_variable = validate_variable_data(variable_data, name, unit)
+    >>> assert validated_variable == var1
+    """
+    items = [item for item in variable_data if item["name"] == name]
+
+    if not items:
+        variable = {"name": name, "unit": unit, "data": np.nan}
+
+    if len(items) == 1:
+        variable = items[0]
+
+    if len(items) > 1:
+        raise RuntimeError(f"Multiple '{name}' exist in '{items}'.")
+
+    if variable["unit"] != unit:
+        raise RuntimeError(
+            f"Unit {unit} does not match the unit {variable['unit']} "
+            f"in {variable} for {name}.")
+
+    return variable
+
+
+def order_data(cubes, obs_names, obs_unit):
+    """Return the data from the cubes in the order defined by ``obs_names``.
+
+    The units from the cubes are checked against ``obs_units``.
+
+    Parameters
+    ----------
+    cubes : :class:`iris.cube.CubeList`
+        The cubes in a random order.
+    obs_names : list
+        The ordered names from the observation files.
+    obs_unit : string
+        The unit of the observation variables.
+
+    Returns
+    -------
+    list
+        The ordered data from the model cubes.
+    """
+    variable_data = []
+    for cube in cubes:
+        variable = {}
+        variable["name"] = cube.name()
+        variable["unit"] = cube.units
+        if np.ma.isMaskedArray(cube.data):
+            variable["data"] = cube.data.data
+        else:
+            variable["data"] = cube.data
+        variable_data.append(variable)
+
+    ordered_model_data = []
+    for obs_name in obs_names:
+        validated_variable = validate_variable_data(variable_data, obs_name,
+                                                    obs_unit)
+        ordered_model_data.append(validated_variable["data"])
+
+    return ordered_model_data
+
+
+def read_text_file(filepath):
+    """Return contents of text file.
+
+    Parameters
+    ----------
+    filepath : string
+        The full path to the text file.
+
+    Returns
+    -------
+    list of dictionaries
+        The contents of the text file where each dictionary corresponds
+        to a line in the file and the key of the dictionary is the name
+        of the column.
+    """
+    contents = []
+    with open(filepath, "r") as file_handle:
+        lines = file_handle.readlines()
+    for line in lines:
+        line_dict = {}
+        items = line.split()
+        line_dict["name"] = items[0]
+        line_dict["unit"] = " ".join(items[1:3]).strip(",")
+        line_dict["data"] = float(items[3])  # float('NaN') == np.nan
```

Comment:
I would change those text files to yaml format and that way you would have your dicts ready done after a a yaml safeload

Reply:
I agree that text files are hardly ideal, but we matched the format of the files used by AutoAssess exactly for this conversion. @alistairsellar, do you think we should update the input file format to remove the need for this additional processing, or do you think they're best kept as is? Perhaps this could be addressed in a future issue?

Reply:
I think it's a good idea to set the standards here, especially if we'll have more of these files added in the future; we can bulk change them at a later stage but that's not ideal. On the other hand, if they are to be interchangeable with Autoassess maybe we should leave them in what Autoassess format they are there. Anyways, am not making too much fuss about it now, am OK to merge like this and we can talk about them next time

Reply:
I've converted the txt files into yaml format and updated the read function accordingly. @valeriupredoi was right - it was pretty much plug and play. The changes can be seen in https://github.com/ESMValGroup/ESMValTool/pull/2282/commits/17426e74d4481979a7de46c40061ebfeb0fd55e3.